src/main/java/frc/robot/utils/CRTGearRatioUtil.java

```diff
@@ -0,0 +1,154 @@
+  private final double secondaryGearRotationRatioTermSimplfied;
+  private final double secondaryGearRotationRatioTerm;
+  private final int maxRotations;
+  private final double mainGearCircumference;
+
+  /**
+   * 
+   * @param mainGearRotationRatioTerm      for every mainGearRotationRatioTerm
+   *                                       rotations of the main gear,
+   *                                       the secondary gear rotates
+   *                                       secondaryGearRotationRatioTerm times
+   * @param secondaryGearRotationRatioTerm for every mainGearRotationRatioTerm
+   *                                       rotations of the main gear,
+   *                                       the secondary gear rotates
+   *                                       secondaryGearRotationRatioTerm times
+   * @param maxRotations                   the maximum number of rotations that
+   *                                       the main gear will make that involve
+   *                                       unique positions of the object
+   *                                       attached to it
+   * @param mainGearCircumference          circumference of the main gear
+   * 
+   * @return Returns optional of the util if the gear ratio works with the number
+   *         of max rotations, otherwise returns an empty optional
+   */
+  public static Optional<CRTGearRatioUtil> init(int mainGearRotationRatioTerm, int secondaryGearRotationRatioTerm,
+      int maxRotations, double mainGearCircumference) {
+    CRTGearRatioUtil u = new CRTGearRatioUtil(mainGearRotationRatioTerm, secondaryGearRotationRatioTerm, maxRotations,
+        mainGearCircumference);
+    if (u.validateGearRatioWithMaxRotations()) { //check if the gear ratio works with the required amount of rotations by the main gear
+      return Optional.of(u);
+    }
+    return Optional.empty();
+  }
+
+  private CRTGearRatioUtil(int mainGearRotationRatioTerm, int secondaryGearRotationRatioTerm, int maxRotations,
+      double mainGearCircumference) {
+    this.mainGearRotationRatioTerm = (double) mainGearRotationRatioTerm;
+    this.secondaryGearRotationRatioTerm = (double) secondaryGearRotationRatioTerm;
+    this.mainGearRotationRatioTermSimplified = (double) (mainGearRotationRatioTerm
+        / gcd(mainGearRotationRatioTerm, secondaryGearRotationRatioTerm));
+    this.secondaryGearRotationRatioTermSimplfied = (double) (secondaryGearRotationRatioTerm
+        / gcd(mainGearRotationRatioTerm, secondaryGearRotationRatioTerm));
+    this.maxRotations = maxRotations;
+    this.mainGearCircumference = mainGearCircumference;
+  }
+
+  //"These two functions speak for themselves" - Hans Niemann commenting his code
+  public static double gcd(double a, double b) {
+    if (b == 0.0)
+      return a;
+    return gcd(b, a % b);
+  }
+
+  public static double mod(double a, double b) {
+    double r = a % b;
+    if (r < 0) {
+      r += b;
+    }
+    return r;
+  }
```

Comment:
Can you use this built-in func? https://stackoverflow.com/a/31140209

Reply:
I would need to cast the doubles to longs and then cast back, do we want to do this?

Reply:
Ah no this is fine